src/main/java/io/reactivex/internal/operators/observable/ObservableDematerialize.java

```diff
@@ -18,9 +18,9 @@
 import io.reactivex.internal.disposables.DisposableHelper;
 import io.reactivex.plugins.RxJavaPlugins;
 
-public final class ObservableDematerialize<T> extends ObservableSource<Try<Optional<T>>, T> {
+public final class ObservableDematerialize<T> extends ObservableSource {
```

Comment:
Makes sense. Does `ObservableFromUpstream` or `ObservableWithUpstream` for that sound good?